### ISSUE TITLE
Preview Window lines up with where you are in code

### ISSUE THREAD

**Issue by RichShumaker** [Jermolene/TiddlyWiki5#1267]
First I love the preview window.
It is cool to change things and be able to check them immediately and test things before committing.
So here is a questions about the Preview Window.

Is there a way for your code and preview window to line up?
When I have a big tiddler the code is below the preview and I have to scroll up and down to check out what I am typing is what is showing up.

What I was envisioning was that the preview would catch up to the start of what line you are on.
So if you go from 
Tag One List to
Tag Two List
The Preview would scroll to the top of the second list bypassing the first list.

Thanks for your help.

Rich Shumaker

**Comment by tobibeer** [comment]
Indeed, much the same as what is needed for WYSIWYG... rather a major beast to tame and surely not really worth it just to have the preview scroll to a given position.

**Comment by twMat** [comment]
Easy for a non-programmer to say, but; hasn't WYSIWYG matured into some kind of "ready made solutions" by now in the programming world? I mean, it must be such a recurring issue for so many software developers so I figure there should be "libraries" or whatever it's called, no?

This question made me just write about an [alternative route](http://tidbits.tiddlyspace.com/#TWYSIWYG) for WYSIWYG in TW.

**Comment by tobibeer** [comment]
> I mean, it must be such a recurring issue for so many software developers so I figure there should be "libraries" or whatever it's called, no?

Sure, for html or rtf or some standard format, yes... with a few glitches... but for a custom parser like that of TiddlyWiki, nope... nothing "standard" about that.

**Comment by twMat** [comment]
Ok, thanks. Off topic, but what must be fulfilled for something to be "easily" integrated into TW? If it is 100% CSS? Html? JS? Or must things always be mangled a lot? Thanks!

**Comment by pmario** [comment]
@Jermolene .. imo should be labeled "feature request"

**Comment by twMat** [comment]
@RichShumaker , @Jermolene 

Here's a site featuring this. Maybe the code is peekable?

https://markdown-it.github.io/

**Comment by RichShumaker** [comment]
Thanks @twMat that is amazing.  Is this a test of a future build?  Just wondering if this was a 'user' modification or if this is a system level fix.  What was the solution? not that I will completely understand it.

**Comment by pmario** [comment]
>https://markdown-it.github.io/ ...

This is not a preview, just an example. 

The markdown-it parser -> rendering mechanism contains some code, that outputs anchors in the html code. ... the tiddlywiki parser has no such mechanism at the moment.

**Comment by twMat** [comment]
@RichShumaker - sorry, that site has nothing to do with TW. I felt it illustrated the idea well and am hoping it is possible to use the code if possible or at least peek at it.

**Comment by RichShumaker** [comment]
d'oh!!! Thanks @twMat  for the visual of what could be done.
